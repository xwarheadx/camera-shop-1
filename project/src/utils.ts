import { PAGE_LIMIT } from './consts';
import { reviewDataType } from '../src/types/review';

export function getPagesCount(cameraCount: number): number {
  return Math.ceil(cameraCount / PAGE_LIMIT);
}

export function getNumeric(page: string) {
  return Number(page.replace( /^\D+/g, ''));
}

export function seperatePrice(priceValue: number) {
  let returnedPrice = String(priceValue);
  if (returnedPrice.length > 3) {
    returnedPrice = `${returnedPrice.slice(0,2)} ${returnedPrice.slice(2)}`;
  }
  return returnedPrice;
}

export function validateInput(target: (EventTarget & HTMLInputElement) | (EventTarget & HTMLTextAreaElement)) {
  if (target.parentElement?.parentElement && target.name === 'rating' && (Number(target.value) > 0)) {
    if (!target.parentElement.parentElement.parentElement?.className.includes('is-valid')) {
      target.parentElement.parentElement.parentElement?.classList.add('is-valid');
      target.parentElement.parentElement.parentElement?.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name !== 'review' && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length > 0) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string'){
    if(target.value.length === 0) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length >= 5) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name === 'review' && target.value.length < 5) {
    if (!target.parentElement.parentElement.className.includes('is-invalid')) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

}
export function validateForm(data: reviewDataType) {
  let validationStatus = true;
  Object.entries(data).forEach(([key, value]) =>{
    if (key !== 'review') {
      if(value === '') {
        const failedInput = document.querySelector(`input[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
        validationStatus = false;
      }
    }
    if(key === 'review') {
      if(String(value).length < 5) {
        const failedInput = document.querySelector(`textarea[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
      }
    }
    if(key === 'rating') {
      if(value < 1) {
        const failedInput = document.querySelector(`input[name=${key}]`);
        if(failedInput && failedInput.parentElement?.parentElement){
          failedInput.parentElement.parentElement.parentElement?.classList.add('is-invalid');
        }
        validationStatus = false;
      }
    }
  });
  return validationStatus;
}


