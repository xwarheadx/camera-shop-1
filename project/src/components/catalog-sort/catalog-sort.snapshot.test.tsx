import { renderWithProviders } from '../../test/utils/render-with-redux';
import CatalogSort from './catalog-sort';


test('Render CatalogSort', () => {
  const { asFragment } = renderWithProviders(<CatalogSort/>);
  expect(asFragment()).toMatchSnapshot();
});
