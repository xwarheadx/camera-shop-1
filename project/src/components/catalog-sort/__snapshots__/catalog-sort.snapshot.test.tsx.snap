// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CatalogSort 1`] = `
<DocumentFragment>
  <div
    class="catalog-sort"
  >
    <form
      action="#"
    >
      <div
        class="catalog-sort__inner"
      >
        <p
          class="title title--h5"
        >
          Сортировать:
        </p>
        <div
          class="catalog-sort__type"
        >
          <div
            class="catalog-sort__btn-text"
          >
            <input
              checked=""
              id="sortPrice"
              name="sort"
              type="radio"
            />
            <label
              for="sortPrice"
            >
              по цене
            </label>
          </div>
          <div
            class="catalog-sort__btn-text"
          >
            <input
              id="sortPopular"
              name="sort"
              type="radio"
            />
            <label
              for="sortPopular"
            >
              по популярности
            </label>
          </div>
        </div>
        <div
          class="catalog-sort__order"
        >
          <div
            class="catalog-sort__btn catalog-sort__btn--up"
          >
            <input
              aria-label="По возрастанию"
              checked=""
              id="up"
              name="sort-icon"
              type="radio"
            />
            <label
              for="up"
            >
              <svg
                aria-hidden="true"
                height="14"
                width="16"
              >
                <use
                  xlink:href="#icon-sort"
                />
              </svg>
            </label>
          </div>
          <div
            class="catalog-sort__btn catalog-sort__btn--down"
          >
            <input
              aria-label="По убыванию"
              id="down"
              name="sort-icon"
              type="radio"
            />
            <label
              for="down"
            >
              <svg
                aria-hidden="true"
                height="14"
                width="16"
              >
                <use
                  xlink:href="#icon-sort"
                />
              </svg>
            </label>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
